src/main/java/io/stargate/sgv2/jsonapi/service/embedding/gateway/EmbeddingGatewayClient.java

```diff
@@ -12,6 +12,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import org.apache.commons.lang3.tuple.Pair;
```

Comment:
I am ok wrt using `Pair` in many cases, but for readability it'd be better to use a Record I think: mostly just so that the first number is clearer (Batch id, not batch size).